### PR TITLE
Migration

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

File: guild-sc/src/user_actions/migration.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        self.require_not_closing();
+
+        let guild_master = self.guild_master().get();
+        let caller = self.blockchain().get_caller();
+        require!(guild_master == caller, "Only guild master may close guild");
+
+        let payments = self.get_non_empty_payments();
+        let mut total_payment = BigUint::zero();
+        for payment in &payments {
+            total_payment += payment.amount;
```
Comment (CostinCarabas) [inline guild-sc/src/user_actions/migration.rs:49]: Check that `payment` has same token_id.

File: guild-sc/src/user_actions/migration.rs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        let total_guild_master_tokens = self.guild_master_tokens().get();
+        require!(
+            total_payment == total_guild_master_tokens.base,
```
Comment (CostinCarabas) [inline guild-sc/src/user_actions/migration.rs:54]: What about the restake rewards?
What if the guild master sends also the rewards?

Comment (dorin-iancu) [reply guild-sc/src/user_actions/migration.rs:54]: Those are sent automatically by unstake.

File: guild-sc/src/user_actions/migration.rs (hunk 1 above)
Comment (dorin-iancu) [reply guild-sc/src/user_actions/migration.rs:49]: Checked automatically in unstake.

**Review by alyn509 — APPROVED** [review]

**Pull request merged**